census_data_aggregator/__init__.py

```diff
@@ -405,3 +406,104 @@ def approximate_percentchange(pair_old, pair_new):
+                * moe (float): The margin of error for n
+        number_replicates (int): number of replicates for simulation, used to estimate margin of error
+        usePareto (logical): use the Pareto distribution to simulate from in upper bin, otherwise use uniform, Pareto is appropriate for income
+
+    Returns:
+        A two-item tuple with the mean followed by the approximated margin of error.
+
+        (774578.4565215431, 128.94103705296743)
+
+    Examples:
+        Estimating the mean for a range of household incomes.
+
+        >>> income = [
+            dict(min=0, max=9999, n=7942251, moe=17662),
+            dict(min=10000, max=14999, n=5768114, moe=16409),
+            dict(min=15000, max=19999, n=5727180, moe=16801),
+            dict(min=20000, max=24999, n=5910725, moe=17864),
+            dict(min=25000, max=29999, n=5619002, moe=16113),
+            dict(min=30000, max=34999, n=5711286, moe=15891),
+            dict(min=35000, max=39999, n=5332778, moe=16488),
+            dict(min=40000, max=44999, n=5354520, moe=15415),
+            dict(min=45000, max=49999, n=4725195, moe=16890),
+            dict(min=50000, max=59999, n=9181800, moe=20965),
+            dict(min=60000, max=74999, n=11818514, moe=30723),
+            dict(min=75000, max=99999, n=14636046, moe=49159),
+            dict(min=100000, max=124999, n=10273788, moe=47842),
+            dict(min=125000, max=149999, n=6428069, moe=37952),
+            dict(min=150000, max=199999, n=6931136, moe=37236),
+            dict(min=200000, max=1000000, n=7465517, moe=42206)
+        ]
+        >>> approximate_mean(income)
+        (98045.44530685373, 194.54892406267754)
+        >>> approximate_mean(income, usePareto=True)
+        (60364.96525340687, 58.60735554621351)
+
+    ... _from the PUMS:
+    https://www2.census.gov/programs-surveys/acs/tech_docs/pums/accuracy/2013_2017AccuracyPUMS.pdf?#
+    """
+
+    # Sort the list
+    range_list.sort(key=lambda x: x['min'])
+
+    idx = len(range_list)
+
+    nb1 = range_list[idx - 2]['n']  # number in second to last bin
+    nb = range_list[idx - 1]['n']  # number in last bin
+    lb1 = range_list[idx - 2]['min']  # lower bound of second to last bin
+    lb = range_list[idx - 1]['min']  # lower bound of last bin
+
+    alpha_hat = (numpy.log(nb1 + nb) - numpy.log(nb)) / (numpy.log(lb) - numpy.log(lb1))
+
+    simulation_results = []
+    for i in range(number_replicates):
+        simulated_values = []
+        simulated_n = []
+        for range_ in range_list[:-1]:
+            se = range_['moe'] / 1.645  # convert moe to se
+            nn = round(numpy.random.normal(range_['n'], se))  # use moe to introduce randomness into number in bin
+            nn = int(nn)  # clean it up
+            simulated_values.append(numpy.random.uniform(range_['min'], range_['max'], size=(1, nn)).sum())  # draw random values within the bin, assume uniform
+            simulated_n.append(nn)
+        if usePareto:
```

Comment:
Add a docstring over this saying something like "A special case to handle the last bin"